packages/p2p/src/components/my-profile/block-user/block-user.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import { observer } from 'mobx-react-lite';
+import { useStores } from 'Stores';
+import { DesktopWrapper, MobileFullPageModal, MobileWrapper } from '@deriv/components';
+import BlockUserModal from 'Components/block-user/block-user-modal';
+import BlockUserTable from 'Components/my-profile/block-user/block-user-table/block-user-table';
+import SearchBox from 'Components/search-box';
+import { my_profile_tabs } from 'Constants/my-profile-tabs';
+import debounce from 'lodash.debounce';
+import { localize } from 'Components/i18next';
+
+const BlockUser = () => {
+    const { general_store, my_profile_store } = useStores();
+
+    const loadBlockedAdvertisers = debounce(search => {
+        my_profile_store.setSearchTerm(search.trim());
+        my_profile_store.loadMoreBlockedAdvertisers();
+    }, 200);
+
+    const onSearch = search => {
+        // Ensures that blocked advertisers list is not reloaded if search term entered is the same
+        if (my_profile_store.search_term !== search.trim()) {
+            my_profile_store.setIsLoading(true);
+            loadBlockedAdvertisers(search);
+        }
+    };
+
+    const onClear = () => {
+        my_profile_store.setSearchTerm('');
+        my_profile_store.setSearchResults([]);
+    };
+
+    return (
+        <React.Fragment>
+            <BlockUserModal
+                advertiser_name={my_profile_store.selected_blocked_user.name}
+                is_advertiser_blocked
+                is_block_user_modal_open={general_store.is_block_user_modal_open}
+                onCancel={() => general_store.setIsBlockUserModalOpen(false)}
+                onSubmit={my_profile_store.onSubmit}
+            />
+            <DesktopWrapper>
+                {my_profile_store.blocked_advertisers_list.length > 0 && (
+                    <SearchBox onClear={onClear} onSearch={onSearch} placeholder={localize('Search')} />
+                )}
+                <BlockUserTable />
```

Comment:
Should we wrap this into a component outside `BlockUser` since it's re-used in `MobileWrapper` as well?

Reply:
Yeap, its duplicated code yikes, will do that asap!